templates/common/_base/files/mtu-migration.yaml

```diff
@@ -0,0 +1,203 @@
+mode: 0755
+path: "/usr/local/bin/mtu-migration.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    set -ex
+
+    run_mtu_migration_phase1() {
+        # In phase1, we copy MCO-generated $mtu_mco_file to $mtu_ongoing_file
+        # and add UUID list to it.
+        # For each selected connection profile:
+        # - find the corresponding device
+        # - trigger an UP event on the device so that the dispatcher script can run its own phase1:
+        #    + source $mtu_ongoing_file
+        #    + set an appropriate MTU on the device and on the routes going through it
+        #    + write mtu_phase2_${iface} file in /tmp
+        # - wait for all mtu_phase2_${iface} files to appear in /tmp and remove $mtu_ongoing_file
+
+        echo "mtu migration phase 1"
+
+        . $mtu_mco_file
+        if [ -z "$TARGET_MTU" ]; then
+            echo "mtu migration error: ${mtu_mco_file} exists but does not contain a TARGET_MTU"
+            cat $mtu_mco_file
+            return
+        fi
+
+        if [ -f $mtu_ongoing_file ]; then
+            echo "mtu migration phase 1: $mtu_ongoing_file already exists. "
+            echo "Is the previous migration being overriden by a new one?"
+            echo "existing ${mtu_ongoing_file}:"
+            cat $mtu_ongoing_file
+            echo "new ${mtu_ongoing_file}:"
+            cat $mtu_mco_file	
+        fi
+
+
+        cp -f $mtu_mco_file $mtu_ongoing_file
+        echo "adding conn_uuids=$conn_uuids to $mtu_ongoing_file"
+        sed -i -e '$aUUIDS='"\"$conn_uuids\"" $mtu_ongoing_file
+
+        for conn_id in $conn_uuids; do
+            echo "processing connection id $conn_id"
+
+            iface=$(nmcli -t -g connection.interface-name connection show $conn_id)
+            echo "iface is $iface"
+
+            if [ -z "$iface" ]; then
+                echo "error: no interface found for connection profile $conn_id: skipping this connection profile"
+                continue
+            fi
+
+            # force UP event on device so that the dispatcher script gets executed
+            echo "bringing down and then up iface=$iface"
+            ip link set "$iface" down
+            ip link set "$iface" up
+
+        done
+    }
+
+
+    run_mtu_migration_phase2() {
```

Comment:
I don't think we need a phase2. At the final stage the interface's MTU is set by however the administrator decided to deploy the MTU changes persistently. And route's MTU do not need to be removed as they do not persist.